### DISCUSSION TITLE
Support for Windows Arm64

### DISCUSSION THREAD

**Discussion by yassinebennani** [jedisct1/libsodium#1360] [Q&A]
Hello,

I could not find a win-arm64 target runtime, is there any plan to support it ?

I forked Sodium.Core repo but I ended with PlatformNotSupportedException 

System.PlatformNotSupportedException : Could not initialize platform-specific components. libsodium-core may not be supported on this platform. See https://github.com/ektrah/libsodium-core/blob/master/INSTALL.md for more information.
  ----> System.DllNotFoundException : Unable to load DLL 'libsodium' or one of its dependencies: The specified module could not be found. (0x8007007E)

Thank you

**Comment by jedisct1** [comment]
It's been supported for a long time, and continuously tested in CI.

Just use Zig:

```sh
zig build -Doptimize=ReleaseFast -Dtarget=aarch64-windows
```

The only reason why it's not included in the Nugget package is that nobody ever confirmed that this actually works.

**Comment by enclave-alistair** [comment]
For what it's worth, we run libsodium on aarch64-windows quite happily (but obviously have to build and then package it ourselves since it's not in the nuget package).

**Comment by prayaas-a** [comment]
With Windows on Arm GitHub runners now available, it should be possible to CI test too: https://github.blog/changelog/2025-04-14-windows-arm64-hosted-runners-now-available-in-public-preview/

**Comment by lazka** [comment]
If would be nice to have a new release with arm64 binaries included, so that pynacl could start using them: https://github.com/pyca/pynacl/issues/775, which in turn blocks pygithub, which is what I'm trying to use :)

**Reply by jedisct1** [reply]
Windows ARM64 precompiled binaries have been available for a long time. https://download.libsodium.org/libsodium/releases/

```sh
$unzip -t libsodium-1.0.20-stable-msvc.zip
Archive:  libsodium-1.0.20-stable-msvc.zip
    testing: libsodium/               OK
    testing: libsodium/ARM64/         OK
    testing: libsodium/ARM64/Debug/   OK
    testing: libsodium/ARM64/Debug/v143/   OK
    testing: libsodium/ARM64/Debug/v143/dynamic/   OK
    testing: libsodium/ARM64/Debug/v143/dynamic/libsodium.dll   OK
    testing: libsodium/ARM64/Debug/v143/dynamic/libsodium.exp   OK
    testing: libsodium/ARM64/Debug/v143/dynamic/libsodium.ilk   OK
    testing: libsodium/ARM64/Debug/v143/dynamic/libsodium.lib   OK
    testing: libsodium/ARM64/Debug/v143/dynamic/libsodium.pdb   OK
    testing: libsodium/ARM64/Debug/v143/ltcg/   OK
    testing: libsodium/ARM64/Debug/v143/ltcg/libsodium.lib   OK
    testing: libsodium/ARM64/Debug/v143/static/   OK
    testing: libsodium/ARM64/Debug/v143/static/libsodium.lib   OK
    testing: libsodium/ARM64/Debug/v143/static/libsodium.pdb   OK
    testing: libsodium/ARM64/Release/   OK
    testing: libsodium/ARM64/Release/v143/   OK
    testing: libsodium/ARM64/Release/v143/dynamic/   OK
    testing: libsodium/ARM64/Release/v143/dynamic/libsodium.dll   OK
    testing: libsodium/ARM64/Release/v143/dynamic/libsodium.exp   OK
    testing: libsodium/ARM64/Release/v143/dynamic/libsodium.lib   OK
    testing: libsodium/ARM64/Release/v143/dynamic/libsodium.pdb   OK
    testing: libsodium/ARM64/Release/v143/ltcg/   OK
    testing: libsodium/ARM64/Release/v143/ltcg/libsodium.lib   OK
    testing: libsodium/ARM64/Release/v143/ltcg/libsodium.pdb   OK
    testing: libsodium/ARM64/Release/v143/static/   OK
    testing: libsodium/ARM64/Release/v143/static/libsodium.lib   OK
    testing: libsodium/ARM64/Release/v143/static/libsodium.pdb   OK
```

**Reply by lazka** [reply]
Ah, indeed; I was confused by the "-stable" having multiple releases and looked at the wrong zip. Sorry! And thanks for the quick response.

**Reply by lazka** [reply]
Random note, could it be that the minisign example on https://download.libsodium.org/libsodium/releases/README.html is missing the "-V" flag?

**Reply by jedisct1** [reply]
> Random note, could it be that the minisign example on https://download.libsodium.org/libsodium/releases/README.html is missing the "-V" flag?

Ah, good catch. Fixed, thanks!

**Reply by lazka** [reply]
Thanks!

arm64 builds seem to work nicely: https://github.com/pyca/pynacl/pull/864

**Comment by hyh19962008** [comment]
I figured out that there is official pre-built ARM64 package only after I built it myself. It's a bit of confusing, since:
1. On [libsodium website](https://doc.libsodium.org/), the introdunction part says Windows only supports x86 and x86_64.
2. On [download page](https://download.libsodium.org/libsodium/releases/), the package comes with only stable-msvc suffix. You can't tell the arch unless you downloaded it.
3. On [Github Release](https://github.com/jedisct1/libsodium/releases), the msvc-stable package is not included here.
